solver_test.rb

```diff
@@ -0,0 +1,32 @@
+require_relative 'solver'
+require 'rspec/expectations'
+
+describe Solver do
+  let(:solver) { Solver.new }
+
+  describe '#factorial' do
+    it 'returns 1 when given 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'returns the correct result for positive integers' do
+      expect(solver.factorial(5)).to eq(120)
+      expect(solver.factorial(10)).to eq(3_628_800)
+    end
+
+    it 'raises an error when given a negative integer' do
+      expect { solver.factorial(-5) }.to raise_error(NonNegativeIntegerError, 'Argument must be a non-negative integer')
+    end
+  end
+
+  describe '#reverse' do
+    it 'reverses a non-empty string' do
+      expect(solver.reverse('hello')).to eq('olleh')
+      expect(solver.reverse('world')).to eq('dlrow')
+    end
+
+    it 'returns an empty string when given an empty string' do
+      expect(solver.reverse('')).to eq('')
+    end
+  end
```

Comment:
- [ ] Great job so far 👏🏻 However, you are missing the `fizzbuzz` method test. Make sure you meet the requirements for this project 👇🏻 And don't forget to start with the tests and method last.

![image](https://user-images.githubusercontent.com/79058364/225659440-550013e9-fc2d-4563-b3a2-051ffcebd182.png)

Reply:
Thank you for the feedback. We will implement the missing methods

Reply:
We have resolved this issue. This can be seen in the images below.

Testing fizzbuzz in solver_spec.rb
![fizzbuzz](https://user-images.githubusercontent.com/32777408/225688958-b74491ac-168e-4b7b-817e-dc7654acc138.png)

fizzbuzz and other methods in solver.rb
![fizzbuzz-solver](https://user-images.githubusercontent.com/32777408/225689185-d271ea70-ad95-4b5b-bd7c-380215e062b1.png)

the test result
![new-rspec-test](https://user-images.githubusercontent.com/32777408/225688744-222391db-dba3-4248-a8e0-3131d510edc6.png)